frontend/src/components/byte-parser/byte-parser.jsx

```diff
@@ -0,0 +1,342 @@
+    }
+
+    /**
+     * Remove the group that was clicked.
+     * 
+     * @param {number} i The group to remove.  
+     */
+    const handleGroupClick = (i) => () => {
+        groups.splice(i, 1);
+        dispatch({
+            type: 'SET_BYTE_PARSER_GROUPS',
+            groups: groups
+        })
+    }
+
+    /**
+     * Render the line, data option selectors, parsed values, and output nodes for the row.
+     * 
+     * @param {number} firstIndexOfRow The first index of a row of bytes, e.g. 0, 8, 16, etc.
+     * @returns {[JSX.Element, JSX.Element, JSX.Element, JSX.Element]} The lines, dropdowns, parsed values, and nodes.
+     */
+    const renderParsedGroups = (firstIndexOfRow) => {
+        const lines = []
+        const typeDropdowns = [];
+        const convertedGroups = [];
+        const outputNodes = [];
+
+        const RIGHT_MOST_X_FRACTION = 1 - (1 / BYTES_IN_ROW / 2)  /* 0.9375 */
+
+        let groupsEndingInThisRow = 0;
+
+        /* iterate each group of bytes and add the line, data options, and parsed values */
+        for (const [groupIndex, {start, end, dataType}] of groups.entries()) {
+            if (!(end >= firstIndexOfRow && end < firstIndexOfRow + BYTES_IN_ROW)) {
+                continue;  /* group doesn't end in this row */
+            }
+
+            groupsEndingInThisRow += 1;
+
+            /* if the group continues from a previous line, set start to the rightmost position */
+            let xStart = Math.max(start, firstIndexOfRow);
+
+            /* put them in 0 -> BYTES_IN_ROW range (we only care about x position in the row) */
+            xStart %= BYTES_IN_ROW;
+            const xEnd = end % BYTES_IN_ROW;
+
+            /* calculate x offset percentage based on position of the group in the row */
+            const x = (RIGHT_MOST_X_FRACTION - ((xEnd + xStart) / 2) / BYTES_IN_ROW) * 100;
+            const y = (ROW_HEIGHT_PX / 2) + ROW_HEIGHT_PX * (groupsEndingInThisRow - 1);
+            
+            lines.push(
+                <Fragment key={groupIndex}>
+                    <line x1={`${x}%`} y1="0" x2={`${x}%`} y2={y} />
+                    <line x1={`${x}%`} y1={y} x2="100%" y2={y} />
+                </Fragment>
+            )
+
+            const groupLength = end - start + 1;  /* the number of 8-bit bytes in the group */
+
+            /* data types where groupLength is divisible by data type size (in bytes) are options */
+            const options = Object.entries(DATA_TYPES_BY_LENGTH).reduce((acc, [size, types]) => (
+                (groupLength % size !== 0) ? acc : [
+                    ...acc,
+                    ...types.map(({ name }) => (
+                        <option key={name} value={name}>{`${name} [${groupLength / size}]`} </option>
+                    ))
+                ]
+            ), [])
+            
+            /* create the data type change handler */
+            const handleTypeChange = (groups, start, end, groupIndex) => (e) => {

```

Comment:
In this case, you can convert it to a single function and just ignore `e`.  You'd then pass the handler to the `select` component like `onChange={() => handleTypeChange(groups, start, end, groupIndex)}`